polyfill/lib/ecmascript.mjs

```diff
@@ -699,11 +712,36 @@ export const ES = ObjectAssign({}, ES2020, {
         millisecond,
         microsecond,
         nanosecond,
-        calendar
-      } = ES.ParseTemporalDateTimeString(ES.ToString(relativeTo)));
+        calendar,
+        ianaName,
+        offset
+      } = ES.ParseISODateTime(ES.ToString(relativeTo), { zoneRequired: false }));
+      if (ianaName) timeZone = ianaName;
       if (!calendar) calendar = GetISO8601Calendar();
       calendar = ES.ToTemporalCalendar(calendar);
     }
+    if (timeZone) {
+      timeZone = ES.ToTemporalTimeZone(timeZone);
+      let offsetNs = null;
+      if (offset) offsetNs = ES.ParseOffsetString(ES.ToString(offset));
+      const epochNanoseconds = ES.InterpretTemporalZonedDateTimeOffset(
+        year,
+        month,
+        day,
+        hour,
+        minute,
+        second,
+        millisecond,
+        microsecond,
+        nanosecond,
+        offsetNs,
+        timeZone,
+        'compatible',
+        'reject'
```

Comment:
We specified that it should work the same as ZonedDateTime.from(), and if the caller wanted different options then they should call from() explicitly with those options. I think it would be confusing to deviate from that.

Reply:
OK sounds reasonable to me.